src/python_framework/nhd_network_utilities.py

```diff
@@ -5,18 +5,50 @@
 import pandas as pd
 import zipfile
 import xarray as xr
-
+import network_dl
 
 def get_geo_file_table_rows(
         geo_file_path = None
+        , data_link = None
         , layer_string = None
         , driver_string = None
         , verbose = False
         , debuglevel = 0
         ):
 
+    if not os.path.exists(geo_file_path):
+    #TODO: Tune this message to report size of download (with bonus: report estimated download time based on a connection test) 
```

Comment:
https://stackoverflow.com/questions/37573483/progress-bar-while-download-file-over-http-with-requests

Reply:
Why don't I go ahead and incorporate that before we finish the pull...

Reply:
Might want to kill your TODO here if it is TODONE